pkgs/development/compilers/openjdk/generic.nix

```diff
@@ -0,0 +1,664 @@
+    };
+
+  ${if atLeast23 then "versionCheckProgram" else null} = "${placeholder "out"}/bin/java";
+
+  ${if !atLeast11 then "doCheck" else null} = false; # fails with "No rule to make target 'y'."
+
+  doInstallCheck = atLeast23;
+
+  ${if atLeast17 then "postPatch" else null} = ''
+    chmod +x configure
+    patchShebangs --build configure
+  '';
+
+  ${if !atLeast17 then "preConfigure" else null} =
+    ''
+      chmod +x configure
+      substituteInPlace configure --replace /bin/bash "${bash}/bin/bash"
+    ''
+    + lib.optionalString (!atLeast11) ''
+      substituteInPlace hotspot/make/linux/adlc_updater --replace /bin/sh "${stdenv.shell}"
+      substituteInPlace hotspot/make/linux/makefiles/dtrace.make --replace /usr/include/sys/sdt.h "/no-such-path"
+    '';
+
+  installPhase =
+    ''
+      mkdir -p $out/lib
+
+      mv build/*/images/${if atLeast11 then "jdk" else "j2sdk-image"} $out/lib/openjdk
+
+      # Remove some broken manpages.
+      rm -rf $out/lib/openjdk/man/ja*
+
+      # Mirror some stuff in top-level.
+      mkdir -p $out/share
+      ln -s $out/lib/openjdk/include $out/include
+      ln -s $out/lib/openjdk/man $out/share/man
+    ''
+    + lib.optionalString atLeast17 ''
+
+      # IDEs use the provided src.zip to navigate the Java codebase (https://github.com/NixOS/nixpkgs/pull/95081)
+    ''
+    + lib.optionalString atLeast11 ''
+      ln -s $out/lib/openjdk/lib/src.zip $out/lib/src.zip
+    ''
+    + ''
+
+      # jni.h expects jni_md.h to be in the header search path.
+      ln -s $out/include/linux/*_md.h $out/include/
+
+      # Remove crap from the installation.
+      rm -rf $out/lib/openjdk/demo${lib.optionalString (!atLeast11) " $out/lib/openjdk/sample"}
+      ${lib.optionalString headless (
+        if atLeast11 then
+          ''
+            rm $out/lib/openjdk/lib/{libjsound,libfontmanager}.so
+          ''
+        else
+          ''
+            rm $out/lib/openjdk/jre/lib/${architecture}/{libjsound,libjsoundalsa,libsplashscreen,libawt*,libfontmanager}.so
+            rm $out/lib/openjdk/jre/bin/policytool
+            rm $out/lib/openjdk/bin/{policytool,appletviewer}
+          ''
+      )}
+    ''
+    + lib.optionalString (!atLeast11) ''
+
+      # Move the JRE to a separate output
+      mkdir -p $jre/lib/openjdk
+      mv $out/lib/openjdk/jre $jre/lib/openjdk/jre
+      mkdir $out/lib/openjdk/jre
+      lndir $jre/lib/openjdk/jre $out/lib/openjdk/jre
+
+      # Make sure cmm/*.pf are not symlinks:
+      # https://youtrack.jetbrains.com/issue/IDEA-147272
+      rm -rf $out/lib/openjdk/jre/lib/cmm
+      ln -s {$jre,$out}/lib/openjdk/jre/lib/cmm
+
+      # Setup fallback fonts
+      ${lib.optionalString (!headless) ''
+        mkdir -p $jre/lib/openjdk/jre/lib/fonts
+        ln -s ${liberation_ttf}/share/fonts/truetype $jre/lib/openjdk/jre/lib/fonts/fallback
+      ''}
+
+      # Remove duplicate binaries.
+      for i in $(cd $out/lib/openjdk/bin && echo *); do
```

Comment:
Wouldn't `ls $out/lib/openjdk/bin` work?

Reply:
That would change the derivation hash.